src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/WorkflowApiClientToInitiateTask.java

```diff
@@ -35,24 +36,17 @@ public WorkflowApiClientToInitiateTask(RestTemplate restTemplate,
     }
 
     @Override
-    public EvaluateDmnResponse<InitiateTaskEvaluateDmnResponse> evaluateDmn(
+    public EvaluateDmnResponse<InitiateEvaluateResponse> evaluateDmn(
         String key,
-        EvaluateDmnRequest<? extends TaskEvaluateDmnRequest> requestParameters
+        EvaluateDmnRequest<? extends EvaluateRequest> requestParameters
     ) {
-        return makePostCall(key, requestParameters).getBody();
-    }
-
-    private ResponseEntity<EvaluateDmnResponse<InitiateTaskEvaluateDmnResponse>> makePostCall(
-        String key,
-        EvaluateDmnRequest<? extends TaskEvaluateDmnRequest> requestParameters
-    ) {
-        return restTemplate.exchange(
+        return restTemplate.<EvaluateDmnResponse<InitiateEvaluateResponse>>exchange(
             String.format("%s/workflow/decision-definition/key/%s/evaluate", workflowApiUrl, key),
             HttpMethod.POST,
             new HttpEntity<>(requestParameters, buildHttpHeaders()),
             new ParameterizedTypeReference<>() {
             }
-        );
+        ).getBody();
```

Comment:
wondering whether this could throw an NPE, I'm not exactly sure :)

Reply:
Yes, we will probably have to handle errors for that. We could leave that for another time once we get more insight into this service and how to react to errors.